### PR TITLE
Filter out snp duplicates

### PR DESCRIPTION
**Pull request by MattCloward**

### CODE REVIEW

File: update_database_scripts/unpackDatabaseCommandLine.R [hunk 1]
```diff
@@ -591,6 +591,27 @@ if (is_ebi_reachable()) {
...
         dplyr::select(-unique_beta_units) # remove the unique beta units column
+      
+      # resolve snps that have differences in rounding which leads to duplicates
+      studyData <- studyData %>%
+        mutate(rounded_beta = floor(betaValue) + signif(betaValue - floor(betaValue), 2)) %>% # create a rounded betaValue column for comparing (2 sig figs after the decimal)
+        mutate(rounded_odds = floor(or_per_copy_number) + signif(or_per_copy_number - floor(or_per_copy_number), 2)) %>% # create a rounded oddsRatio column for comparing (2 sig figs after the decimal)
+        group_by(variant_id, risk_allele, pvalue, trait, pvalue_description, beta_description, beta_unit, ogValueTypes, studyID, rounded_beta, rounded_odds) %>%
+        mutate(num_repeated_snps = dplyr::n()) %>%
+        ungroup() %>%
+        mutate(or_per_copy_number = ifelse(num_repeated_snps > 1, rounded_odds, or_per_copy_number)) %>%
```
Comment (mpage21) [inline update_database_scripts/unpackDatabaseCommandLine.R:602]: So if there are more than one snp, we are keeping the rounded_odds value?

File: update_database_scripts/unpackDatabaseCommandLine.R [hunk 2]
```diff
@@ -591,6 +591,27 @@ if (is_ebi_reachable()) {
...
+        distinct(variant_id, risk_allele, pvalue, trait, pvalue_description, beta_description, beta_unit, ogValueTypes, studyID, rounded_beta, rounded_odds, .keep_all = TRUE) %>%
+        dplyr::select(-rounded_beta, -rounded_odds, -num_repeated_snps) # remove the rounded columns and num_repeated_snps column
+      
+      # remove the duplicate snps that we can't resolve
+      studyData <- studyData %>%
+        group_by(variant_id, risk_allele, trait, pvalue_description, beta_description, ogValueTypes, studyID) %>%
+        mutate(num_repeated_snps = dplyr::n()) %>%
+        ungroup() %>%
+        group_by(trait, pvalue_description, beta_description, ogValueTypes, studyID) %>%
+        filter(num_repeated_snps==1) %>%
```
Comment (mpage21) [inline update_database_scripts/unpackDatabaseCommandLine.R:613]: Does this get rid of ALL the snps from a study (by the unique key of trait, pValueAnnotation, betaAnnotation, ogValueTypes, studyID) or just the duplicated ones? We need to make sure that ALL the snps from that combination are removed.

File: update_database_scripts/unpackDatabaseCommandLine.R (hunk 1 above)
Comment (MattCloward) [reply update_database_scripts/unpackDatabaseCommandLine.R:602]: That is what Justin suggested.

**Review by mpage21 — APPROVED** [review]

File: update_database_scripts/unpackDatabaseCommandLine.R (hunk 2 above)
Comment (MattCloward) [reply update_database_scripts/unpackDatabaseCommandLine.R]: Fixed. The whole study is now removed.

**Pull request merged**